syntax = "proto3";

option java_multiple_files = true;

message ComposePackage {
    bytes compose_file = 1;
}

message ComposeIdentifier {
    string compose_id = 1;
}

message Empty {

}

message ResourceGroupCompose {

    message PoPCompose {
        string name = 1;
        string interfaceEndpoint = 2;
    }

    message MetadataEntryCompose {
        string key = 1;
        string value = 2;
    }

    message NetworkCompose {
        string name = 1;
        string poPName = 2;
        string cidr = 3;
        string networkId = 4;
    }

    message VDUCompose {
        string name = 1;
        string imageName = 2;
        string netName = 3;
        string poPName = 4;
        string computeId = 5;
        string ip = 6;
        repeated MetadataEntryCompose metadata = 7;
    }

    string name = 1;
    repeated PoPCompose pops = 2;
    repeated NetworkCompose networks = 3;
    repeated VDUCompose vdus = 4;
}

service ComposeHandler {

    rpc UpCompose(ComposePackage) returns (ResourceGroupCompose) {}

    rpc RemoveCompose(ComposeIdentifier) returns (Empty) {}
}
